# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceStorageWorker.cpp
# Commit: 32515f9b41c7
# Full Hash: 32515f9b41c77d6ccd5831eed28953037c4fb319
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 2
# Description:
#   Bug 1800659 - P2 remove nsICancelableRunnable inheriting form WorkerRunnable. r=asuth
#   
#   This patch is only remove the inheriting, but keeping Cancel() implmentation.
#   The Cancel() is needed in some special cases, such as Worker enters Canceling when WorkerScope is not created.
#   
# ==============================================================================

diff -r 910dabb8545d -r 32515f9b41c7 dom/performance/PerformanceStorageWorker.cpp
--- a/dom/performance/PerformanceStorageWorker.cpp	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/performance/PerformanceStorageWorker.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -50,7 +50,7 @@
 
   nsresult Cancel() override {
     mStorage->ShutdownOnWorker();
-    return WorkerRunnable::Cancel();
+    return NS_OK;
   }
 
   bool PreDispatch(WorkerPrivate* aWorkerPrivate) override { return true; }